src/frontend/next/src/components/Posts/index.tsx

```diff
@@ -1,3 +1,99 @@
-const Posts = () => <h3>Posts Component</h3>;
+import { useState, useEffect } from 'react';
+import { useSWRInfinite } from 'swr';
+import { Container, createStyles, Grid } from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import { usePrevious } from 'react-use';
+import SentimentDissatisfiedRoundedIcon from '@material-ui/icons/SentimentDissatisfiedRounded';
+import Timeline from './Timeline';
+import useSiteMetaData from '../../hooks/use-site-metadata';
+import useFaviconBadge from '../../hooks/use-favicon-badge';
+import { Post } from '../../interfaces';
+
+const useStyles = makeStyles(() =>
+  createStyles({
+    root: {
+      padding: 0,
+      maxWidth: '785px',
+    },
+    error: {
+      color: '#B5B5B5',
+      fontFamily: 'Roboto',
+      fontSize: '5rem',
+      paddingBottom: '30px',
+    },
+    errorIcon: {
+      color: '#B5B5B5',
+      fontSize: '10rem',
+      paddingBottom: 0,
+    },
+  })
+);
+
+// Refresh post data every 5 mins
+const REFRESH_INTERVAL = 5 * 60 * 1000;
+
+const Posts = () => {
+  const classes = useStyles();
+  const { telescopeUrl } = useSiteMetaData();
+  const [currentPostId, setCurrentPostId] = useState<string | null>();
+
+  const { data, size, setSize, error } = useSWRInfinite<Post[]>(
+    (index: number) => `${telescopeUrl}/posts?page=${index + 1}`,
```

Comment:
When I load https://deploy-preview-1504--telescope-next.netlify.app/ it looks like `telescopeUrl` is undefined on Netlify, it is loading this URL:

https://deploy-preview-1504--telescope-next.netlify.app/undefined/posts?page=1

Filed #1589 to deal with this.